scripts/xtensa-build-zephyr.py

```diff
@@ -303,14 +274,31 @@ def check_west_installation():
 	print(f"Found west: {west_path}")
 
 def find_west_workspace():
-	"""[summary] Scans for west workspaces using west topdir command by checking script execution
-	west_top and SOF_TOP directories.
+	"""[summary] Validates whether west workspace had been initialized and points to SOF manifest.
 
-	:return: [description] Returns path when west workspace had been found, otherwise None
+	:return: [description] Returns path when west workspace had been found and path of the manifest
+	to which west is initialized to, otherwise None, None.
 	:rtype: [type] bool, pathlib.Path
 	"""
-	global west_top, SOF_TOP
-	paths = [ pathlib.Path(os.getcwd()), SOF_TOP, west_top]
+	global west_top, SOF_TOP, west_manifest_path
+	result_rootdir = execute_command(["west", "topdir"], capture_output=True, cwd=west_top,
+		timeout=10, check=False)
+	if result_rootdir.returncode != 0:
+		return None, None
+	west_root_dir = pathlib.Path(result_rootdir.stdout.decode().strip('\n')).resolve()
+	result_manifest_dir = execute_command(["west", "config", "manifest.path"], capture_output=True,
+		cwd=west_top, timeout=10, check=True)
+	west_manifest_dir = pathlib.Path(west_root_dir, result_manifest_dir.stdout.decode().strip('\n')).resolve()
+	manifest_file_result = execute_command(["west", "config", "manifest.file"], capture_output=True,
+		cwd=west_top, timeout=10, check=True)
+	returned_manifest_path = pathlib.Path(west_manifest_dir, manifest_file_result.stdout.decode().strip('\n'))
+	west_workspace_dir = pathlib.Path(west_root_dir, ".west")
```

Comment:
Hmmm, yes it is implementation detail.
The case in this function is that end-user may have initialized west in right west root dir but to other manifest than SOFs (eg. Zephyr manifest). In this case, I wanted to stop script processing with informative message on what to do.
End-user that knows west tool will know that running `west init -l <SOF_TOP>` will do the job and reinitialize west to proper manifest. However it seemed to me that deleting hidden .west directory is a simpler instruction to unexperienced user. And that is why this Path is constructed - to be displayed in error message.

How can we do it differently in this scenario?
We can change error message to contain description like:
"Please in <west_root> directory run `west init -l <SOF_TOP>` command.
But I think I will go even further with this so the user does not have to restart the script.
I will add a interactive prompt just asking for permission to reinitialize west to SOFs manifest.

Reply:
> > .west/ is an internal implementation detail, it should never be useful.

> end-user may have initialized west in right west root dir but to other manifest than SOFs (eg. Zephyr manifest).
> End-user who knows west tool will know that running west init -l <SOF_TOP> will do the job and reinitialize west to proper manifest. However it seemed to me that deleting hidden .west directory is a simpler instruction to unexperienced user.

I understand now, thanks.

@mbolivar-nordic what is the recommended way to "undo" a west workspace while keeping git repo(s) in place? For the purpose of (re-)initializing with another manifest instead in the same place. west's documentation is outstanding but I don't remember anything about "de-initialization". Of course it's always possible to move desired repos out of the way before deleting the entire workspace but that method lacks... refinement?

Reply:
The only way currently is to remove the .west directory. However, instead of hard-coding the location, you may wish to consider using this API to find it: https://docs.zephyrproject.org/latest/develop/west/west-apis.html#west.util.west_dir

There's no explicit "de-" or "re-" initialization mechanism. We tried to design one during west's early days, and got into a bunch of internal arguments about what to do, then abandoned the idea until a compelling use case came along that would justify bringing it in again. Is this that time? If so, please file an issue.

Reply:
Thanks @mbolivar-nordic , I just filed new
- https://github.com/zephyrproject-rtos/west/issues/593